### PR TITLE
Added time row for collection camp

### PR DESCRIPTION
**Pull request by nishant22029**
## Summary by CodeRabbit

- **New Features**
	- Standardized date and time formatting across the 'collection-camp' and 'dropping-center' targets.
	- Introduced a new row to clearly display the time range between start and end dates.
	- Added a shortcode for an office visit contribution button that generates a volunteer signup option.
	- Registered a shortcode for the collection camp landing page for use in posts or pages.
	- Enhanced query variables for better URL parameter handling.
	- Implemented redirection logic after individual creation based on the source of the individual.

- **Bug Fixes**
	- Enhanced clarity of displayed information by simplifying date presentation and distinguishing the time period.

### CODE REVIEW

File: wp-content/plugins/goonj-blocks/src/render.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
  * @see https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#render
  */
 
+define('DATE_FORMAT', 'd-m-Y');
+define('TIME_FORMAT', 'h:i A');
```
Comment (pokhiii) [inline wp-content/plugins/goonj-blocks/src/render.php:7]: Remove these – these names suggests that we are setting global date and time format.

File: wp-content/plugins/goonj-blocks/src/render.php [hunk 2]
```diff
@@ -39,6 +41,14 @@
 	$action_target['id']
 );
 
+function format_date($date) {
+    return $date->format(DATE_FORMAT);
+}
+
+function format_time_range($start_date, $end_date) {
+    return $start_date->format(TIME_FORMAT) . ' - ' . $end_date->format(TIME_FORMAT);
+}
```
Comment (pokhiii) [inline wp-content/plugins/goonj-blocks/src/render.php:50]: let's create a new file `functions.php` inside this plugin and move these function there.

```suggestion
require_once __DIR__ . '/functions.php';
```

File: wp-content/plugins/goonj-blocks/src/render.php [hunk 3]
```diff
@@ -39,6 +41,14 @@
 	$action_target['id']
 );
 
+function format_date($date) {
```
Comment (pokhiii) [inline wp-content/plugins/goonj-blocks/src/render.php:44]: These are global function so we should always prefix with plugin slug
```suggestion
function gb_format_date($date) {
```

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/plugins/goonj-blocks/src/render.php (hunk 1 above)
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/render.php]: sure

File: wp-content/plugins/goonj-blocks/src/render.php (hunk 2 above)
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/render.php]: sure

File: wp-content/plugins/goonj-blocks/src/render.php (hunk 3 above)
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/render.php]: sure

File: wp-content/themes/goonj-crm/functions.php [hunk 4]
```diff
@@ -495,6 +495,64 @@ function goonj_contribution_volunteer_signup_button() {
...
+
+        $contact = \Civi\Api4\Contact::get(FALSE)
+            ->addSelect('contact_sub_type')
+            ->addWhere('id', '=', $individualId)
+            ->execute()
+            ->first();
+
+		if (empty($contact)) {
+			\Civi::log()->info('Contact not found', ['contact' => $contact['id']]);
+			return;
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: removed file changes from this PR

File: wp-content/plugins/goonj-blocks/src/functions.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+function gb_format_date($date) {
+    return $date->format('d-m-Y');
+}
+
+function gb_format_time_range($start_date, $end_date) {
+    return $start_date->format('h:i A') . ' - ' . $end_date->format('h:i A');
+}
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/functions.php]: done changes

File: wp-content/plugins/goonj-blocks/src/functions.php [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+function gb_format_date($date) {
+    return $date->format('d-m-Y');
+}
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/functions.php]: done changes

**Review by pokhiii — APPROVED** [review]

**Pull request merged**